core/state/statedb.go

```diff
@@ -1274,7 +1289,7 @@ func (s *StateDB) Commit(block uint64, deleteEmptyObjects bool) (common.Hash, er
 	if origin == (common.Hash{}) {
 		origin = types.EmptyRootHash
 	}
-	if root != origin {
```

Comment:
why did we remove this?

Reply:
Due to some consecutive blocks having the same root, I agree that we can probably hoist this to the caller instead, and probably drop the `UpdateSameRoot`